docs/validators/key-management/multikey-nodes.md

```diff
@@ -0,0 +1,123 @@
+---
+id: multikey-nodes
+title: Multikey nodes management
+---
+
+[comment]: # (mx-abstract)
+This page contains information about how to manage multiple keys on an observing squad.
+
+[comment]: # (mx-context-auto)
+## Multikey architecture overview
+
+At the mainnet launch, and up until the release candidate RC/v1.6.0 becomes mainstream, each node could have managed only
+one key. So the relationship between the nodes and keys is `1:1`, the node "following" the shard where the managed key is assigned.
+The main idea behind the multikey feature is that a set of keys can be managed by an observing squad set. The same set of keys
+should be provided for all observing squad nodes, so the nodes can **pick** any key from that set and use it to propose or sign
+blocks whenever the consensus group in a round contains at least one managed key. The observing node that runs in multikey mode
+is able to use the provided keys in any combination possible. So one node can use one key to propose a block and one or more other
+keys to already sign on that proposed block automatically to improve performance.
+So, with the multikey feature, the relationship now becomes `n:[num_shards]`, providing that `n` is the number of keys 
+managed by an entity.
+
+:::important
+This feature is purely optional. Normal `1:1` relationship between the keys and the nodes is still supported. The multikey
+mode should optimize the costs when running a set of keys (check [Economics running multikey nodes](#economics-running-multikey-nodes) section)
+:::
+
+[comment]: # (mx-context-auto)
+## General implementation details
+
+The observers running with the multikey feature, beside deciding the consensus group (which is normally done on each node), 
+can access the provided keys set and use, in any combination one or more key, if the node detects that the at least one managed
+key is part of the consensus group.
+Code-wise, the implementation focused on the `consensus`, `keyManagement` and `heartbeat` packages.
+
+[comment]: # (mx-context-auto)
+### Heartbeat information
+
+The squad managing the set of keys (we will call them multikey nodes or multikey squad), will pass the validators BLS information tight to 
+"virtual" peer IDs. A "virtual peer ID" is a generated p2p identity that the p2p network can not connect to as it does not
+have a real address bind to. Consequentially, this feature brings a new layer of security as the multikey nodes interface 
+the network on a random BLS key, emitting managed heartbeat information on the behalf of generated p2p identities. 
```

Comment:
the multikey node interfaces the network using a PeerID (P2P key) not the random BLS key, the network might not even know the observer BLS key.

Reply:
rephrased